crates/uv/tests/branching_urls.rs

```diff
@@ -232,8 +232,8 @@ fn root_package_splits_transitive_too() -> Result<()> {
         "python_version < '3.12'",
     ]
     dependencies = [
-        { name = "idna" },
-        { name = "sniffio" },
+        { name = "idna", marker = "python_version < '3.12'" },
+        { name = "sniffio", marker = "python_version < '3.12'" },
```

Comment:
I think these are superfluous because the dependency on `anyio` from `a` above is already gated on marker expressions.

(I think this change overall brings us closer to "write out the full marker expressions for each distribution" that could potentially obviate the need for a DAG traversal.)